sklearn/cluster/birch.py

```diff
@@ -537,16 +537,11 @@ def partial_fit(self, X=None, y=None):
             return self._fit(X)
 
     def _check_fit(self, X):
-        is_fitted = hasattr(self, 'subcluster_centers_')
+        check_is_fitted(self, ['subcluster_centers_', 'partial_fit_'],
+                        all_or_any=any)
 
-        # Called by partial_fit, before fitting.
-        has_partial_fit = hasattr(self, 'partial_fit_')
-
-        # Should raise an error if one does not fit before predicting.
-        if not (is_fitted or has_partial_fit):
-            raise NotFittedError("Fit training data before predicting")
-
-        if is_fitted and X.shape[1] != self.subcluster_centers_.shape[1]:
+        if hasattr(self, 'subcluster_centers_') and X.shape[1] != \
```

Comment:
We already know that it has `subcluster_centers_` otherwise the `check_is_fitted` would have already raised an error.

```python
if X.shape[1] != self.subcluster_centers_.shape[1]:
    ...
```

Reply:
The `check_is_fitted` is called with two arguments `'subcluster_centers_` and `partial_fit_` and `all_or_any` is set to `any`, so it is not always the case that the object has `subcluster_centers_ `. To convince yourself, remove the `hasattr(self, 'subcluster_centers_')` and you will see that one of the tests fails.

Reply:
Oh ok, I missed that. Could you then remove the `\` and use parentheses instead.